pages/home/home.sql

```diff
@@ -1,3 +1,20 @@
+-- BLOCK insert_xc101_viewer_if_has_course
+WITH
+example_course AS (
+    SELECT * FROM pl_courses WHERE is_example_course IS true
+)
+INSERT INTO course_permissions (user_id, course_id, course_role)
+    SELECT
+        cp.user_id, xc.id, 'Viewer'
+    FROM
+        course_permissions AS cp
+        JOIN example_course AS xc ON (xc.id != cp.course_id)
+    WHERE
+        cp.user_id = $user_id
+        AND cp.course_role IN ('Owner', 'Editor', 'Viewer')
+    LIMIT 1
+ON CONFLICT DO NOTHING
```

Comment:
```suggestion
INSERT INTO course_permissions (user_id, course_id, course_role)
    SELECT
        $user_id, c.id, 'Viewer'
    FROM
        pl_courses AS c
    WHERE
        c.is_example_course
        AND EXISTS ( -- is the user at least a Viewer in some course?
            SELECT 1 FROM course_permissions
            WHERE cp.user_id = $user_id AND cp.course_role IN ('Owner', 'Editor', 'Viewer')
        )
ON CONFLICT DO NOTHING
```

This change works even if multiple courses are flagged as an example course (for future-proofing). It doesn't bother to check whether the existing Viewer perms are in the example course or not, because it doesn't matter.